inject/src/main/java/io/micronaut/inject/qualifiers/NameQualifier.java

```diff
@@ -66,7 +65,7 @@
                     }
                     return candidate.getBeanType().getSimpleName();
                 });
-                return typeName.equalsIgnoreCase(name) || typeName.toLowerCase(Locale.ENGLISH).startsWith(name);
+                return typeName.equalsIgnoreCase(name) || typeName.equalsIgnoreCase(name + beanType.getSimpleName());
```

Comment:
I also looked into this issue and I'm wondering what is the use-case for the second test `name+beanType` ?

Reply:
@posto It's part of the JSR-330 spec

Reply:
https://github.com/javax-inject/javax-inject/blob/master/tck/org/atinject/tck/auto/Convertible.java#L52

That line expects a class named `SpareTire` to be injected